File: LeetCode/LeetCode Practice Problems/125- Valid Palindrome.cpp
class Solution {
public:
    void increment(int& low,int& high){
        low++;
        high--;
    }
    
    bool isAlphaNum(char c){
        return (isdigit(c) || isalpha(c));
    }
    bool isPalindrome(string s) {
        if(s.length() == 0)
            return true;
        int low = 0 , high = s.length() - 1;
        
        while(low < high){
            if(s[low] == s[high])
            {
                increment(low,high);
                continue;
            }
            
            if(tolower(s[low]) == s[high] || s[low] == tolower(s[high]))
            {
                increment(low , high);
                continue;
            }
            
            if(!isAlphaNum(s[low]))
            {
                low++;
                continue;
            }
            else if(!isAlphaNum(s[high]))
            {
                high--;
                continue;
            }
            
            return false;
        }
        
        return true;
    }
};

File: LeetCode/LeetCode Practice Problems/125-Valid Palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String st = "";
        s = s.toLowerCase();
        
        for(char ch: s.toCharArray()) {
            if((ch >='a' && ch<='z') || (ch >='0' && ch <= '9')) {
                st = st + ch;
            }
        }
        
        char c[] = st.toCharArray();
        int i = 0 , j = c.length - 1;
        while(i < j) {
            if(c[i] != c[j]) {
                return false;
            }
            i++;
            j--;
        }
        
        return true;
    }
}
